File: reporting-dashboard/features/report/types/report.ts
export interface PreviewRequest {
  report_type: string;
  start_date: string;
  end_date: string;
  channel?: string;
  brand?: string;
  main_category?: string;
}

export interface ExportRequest {
  report_type: string;

  start_date: string;
  end_date: string;

  divisi: string;
  departemen: string;
  customer: string;

  nama_layanan: string;
  nama_sub_layanan: string;

  layanan_cc_non_cc: string;

  segment: string;
  sub_segment: string;

  kota: string;

  channel?: string;
  brand?: string;
  main_category?: string;
}

export interface ReportOptions {
  report_types: {
    label: string;
    value: string;
  }[];

  channels: string[];

  brands: string[];

  main_categories: string[];
}

export type PreviewRow = Record<string, string | number | boolean | null>;

export type PreviewResponse = PreviewRow[];

File: reporting-backend/app/schemas/report.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


# ============================================================
# REPORT PREVIEW
# ============================================================

class PreviewRequest(BaseModel):
    report_type: str

    channel: Optional[str] = ""
    brand: Optional[str] = ""
    main_category: Optional[str] = ""

    start_date: date
    end_date: date


# ============================================================
# REPORT EXPORT
# ============================================================

class ExportRequest(BaseModel):
    report_type: str

    channel: Optional[str] = ""
    brand: Optional[str] = ""
    main_category: Optional[str] = ""

    start_date: date
    end_date: date

    divisi: Optional[str] = ""
    departemen: Optional[str] = ""
    customer: Optional[str] = ""

    nama_layanan: Optional[str] = ""
    nama_sub_layanan: Optional[str] = ""
    layanan_cc_non_cc: Optional[str] = ""

    segment: Optional[str] = ""
    sub_segment: Optional[str] = ""
    kota: Optional[str] = ""